src/__tests__/components/speedButton/speedButton.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { render, fireEvent } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+
+import SpeedButton from '../../../components/speedButton';
+import speedButtonProps from '../../../utils/speedButtonsProps';
+
+const [slowerButtonProps, defaultButtonProps, fasterButtonProps] = speedButtonProps;
+
+describe('SpeedButton',() => {
+  describe('when clicked',() => {
+    describe('in the slower button',() => {
+      it('set the time to double', () => {
+        let initialTime = 1000;
+
+        const button = (
+          <SpeedButton
+            onClick={(time) => initialTime = time}
+            time={initialTime}
+            {...slowerButtonProps}
+          />
+        );
+
+        const { getByText } = render(button);
+
+        fireEvent.click(getByText('Slower'));
+
+        expect(initialTime).toBe(2000);
```

Comment:
I think you could use `jest.fn()` here to detect the click event:
```js
it("set the time to double", () => {
  const onClick = jest.fn()
  const button = (
    <SpeedButton
      onClick={onClick}
      time={initialTime}
      {...slowerButtonProps}
    />
  );

  const { getByText } = render(button);

  fireEvent.click(getByText("Slower"));

  expect(onClick).toHaveBeenCalledWith(2000);
});
```

This way, you make sure the event `onClick` is been fired and also the value changes with `toHaveBeenCalledWith` method 😄 

Also, you can improve a little bit this test creating a method to render the component:
```js
const renderComponent = (props = {}) => render(<SpeedButton {...props} />);
```
Then, you can write you test like this:
```js
it("set the time to double", () => {
  const onClick = jest.fn();
  const { getByText } = renderComponent({
    onClick,
    time: initialTime,
    ...slowerButtonProps
  });

  fireEvent.click(getByText("Slower"));

  expect(onClick).toHaveBeenCalledWith(2000);
});
```